src/Compilers/CSharp/Portable/Lowering/ExtensionMethodBodyRewriter.cs

```diff
@@ -58,13 +55,20 @@ public ExtensionMethodBodyRewriter(MethodSymbol sourceMethod, SourceExtensionImp
 
             Debug.Assert(symbol.Parameters.Length == rewrittenParameters.Length);
 
-            if (!rewrittenParameters.IsEmpty)
+            if (symbol.MethodKind is MethodKind.LambdaMethod || !rewrittenParameters.IsEmpty)
             {
                 var builder = _symbolMap.ToBuilder();
+                if (symbol.MethodKind is MethodKind.LambdaMethod)
+                {
+                    // BoundMethodDefIndex in instrumentation will refer to the lambda method symbol, so we need to map it.
+                    builder.Add(symbol, rewritten);
```

Comment:
>builder.Add(symbol, rewritten); [](http://example.com/codeflow?start=20&length=31)

I think this belongs in VisitLambda method instead.

Reply:
Thanks okay. I was trying to avoid extra allocations